shared/teams/new-team/wizard/new-team-info.tsx

```diff
@@ -9,15 +9,34 @@ import {pluralize} from '../../../util/string'
 import {InlineDropdown} from '../../../common-adapters/dropdown'
 import {FloatingRolePicker} from '../../role-picker'
 import * as RouteTreeGen from '../../../actions/route-tree-gen'
+import * as RPCTypes from '../../../constants/types/rpc-gen'
 
 const NewTeamInfo = () => {
   const dispatch = Container.useDispatch()
   const nav = Container.useSafeNavigation()
 
   const teamWizardState = Container.useSelector(state => state.teams.newTeamWizard)
-  const teamNameTaken = false // TODO: get this live (Y2K-1524 and probably useRPC)
 
   const [name, setName] = React.useState(teamWizardState.name)
+  const [teamNameTakenError, setTeamNameTakenError] = React.useState<string | null>(null)
+  const [teamNameTaken, setTeamNameTaken] = React.useState(false)
+  const checkTeamNameTaken = Container.useRPC(RPCTypes.teamsUntrustedTeamExistsRpcPromise)
+  React.useEffect(() => {
```

Comment:
please make sure we are debouncing heavily on this, this endpoint has to do 3 queries IIRC and it would be bad for it to fire every keystroke

Reply:
also should be single-flought (? is that a word)

Reply:
agree with the floughtiness but we don't necessarily need to debounce too much I think, it's not happening that often

Reply:
can we cache results for team names that have been tried?

Reply:
It seems we do need to debounce here. Otherwise this thing gets fired on every key press in the input.

Reply:
Now debouncing at 100ms. I talked with @heronhaye and he suggested we should do the single-flightedness in go, but that maybe we don't need to do it yet until we see how much load actually comes from this.